upup/pkg/fi/http.go

```diff
@@ -67,6 +70,57 @@ func DownloadURL(url string, dest string, hash *hashing.Hash) (*hashing.Hash, er
 	return hash, nil
 }
 
+func addAuthHeadersFromEnvironment(client *http.Client, req *http.Request) error {
```

Comment:
It _might_ be easier to pass the URL to nodeup etc as a `gs://` URL, rather than rewriting to storage.googleapis.com.  I _think_ that nodeup will then know to get a GCP token etc

Reply:
Unforunately, on flatcar, it doesn't seem to work - gsutil requires `docker` on flatcar, which isn't up yet at this point.  I can try to work around that if this is important.

Reply:
Ah, I see that it works on your config - maybe I can just make it work with mine somehow!  I'll take a look...